packages/context/src/inject.ts

```diff
@@ -293,6 +336,12 @@ function resolveAsGetter(
 }
 
 function resolveAsSetter(ctx: Context, injection: Injection) {
+  const targetType = inspectTargetType(injection);
+  if (targetType && targetType !== Function) {
+    throw new Error(
+      `The target type ${targetType.name} is not a Setter function`,
```

Comment:
Ditto: When the user receives this error, how can they find the place that has triggered it?